src/ecdsa/ellipticcurve.py

```diff
@@ -185,6 +185,15 @@ def __init__(self, curve, x, y, z, order=None, generator=False):
                 doubler = doubler.double().scale()
                 self.__precompute.append((doubler.x(), doubler.y()))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
```

Comment:
the RWLock is used to ensure that that the object is self-consistent, this copy should use it

Reply:
Done. If it can't acquire the lock, I have it return None. Think that's fine or prefer something else?